### PR TITLE
#161348759 Rating an article

### PR DESCRIPTION
**Pull request by judeinno**
#### What does this PR do?

- A user is able to see the rating of an article and also rate the article.

#### Description of Task to be completed?

- Add radio buttons for the user to be able to rate an article.
- Add 5-star rating to show the article rating.

#### How should this be manually tested?

- Click any rating on the radio buttons and click submit.
- The when a user gets an article by ID the rating is returned in 5-star rating format.

#### What are the relevant pivotal tracker stories?

[#161348759](https://www.pivotaltracker.com/story/show/161348759)

#### Screenshots 

image

### CODE REVIEW

File: src/actions/loginActions/loginAction.js [hunk 1]
```diff
@@ -23,13 +20,8 @@ export const loginThunk = data => (dispatch) => {
   return axios.post('https://ah-backend-thanos-staging.herokuapp.com/api/users/login', userdata)
     .then((res) => {
       dispatch(loginSuccess(res.data.results));
```
Comment (dnuwa) [inline src/actions/loginActions/loginAction.js]: this is good

**Review by dnuwa — DISMISSED** [review]

**Review by brucemakallan — DISMISSED** [review]
Noticed that you removed `swal` from the actions (because they should not have side effects). PR looks good.

File: src/actions/articleActions.js [hunk 2]
```diff
@@ -55,10 +55,7 @@ export const getLikeStatusThunk = ({ articleId, token }) => (dispatch, getState)
   const url = `${APP_URL}/articles/${articleId}/like_status`;
   return axios.get(url, { headers: { Authorization: `Token ${token}` } })
     .then((response) => {
-      const obj = response.data.results.filter(
-        res => res.user.username === localStorage.getItem('username')
-        && res.article === getState().articleReducer.article.id,
-      );
+      const obj = response.data.results.filter(res => res.user.username === localStorage.getItem('username') && res.article === getState().articleReducer.article.id);
```
Comment (tibetegya) [inline src/actions/articleActions.js]: this line is too long.

File: src/reducers/reducers.test.js [hunk 3]
```diff
@@ -3,21 +3,22 @@ import combinedReducers from './index';
...
   it('should return the initial state for user reducer', () => {
     expect(userReducer(undefined, {})).toEqual(initialState.userReducer);
   });
 
   it('should return the initial state for articles reducer', () => {
     expect(articleReducer(undefined, {})).toEqual(initialState.articleReducer);
   });
+
+  it('should return the initial state for user reducer', () => {
+    expect(ratingReducer(undefined, {})).toEqual({});
```
Comment (tibetegya) [inline src/reducers/reducers.test.js]: Why are you passing undefined to this test?

Comment (judeinno) [reply src/reducers/reducers.test.js]: We don't want the state to be defined so that it returns the initial state.

**Review by brucemakallan — DISMISSED** [review]
Thanks for fixing the tests and code climate issues. Looks good.

**Review by RachaelNantale — DISMISSED** [review]
Looks Good, The UI is well implemented

**Review by dnuwa — DISMISSED** [review]

**Review by Sulaiman-Mozes — DISMISSED** [review]

**Review by Sulaiman-Mozes — APPROVED** [review]

**Review by brucemakallan — APPROVED** [review]

**Review by dnuwa — APPROVED** [review]

**Pull request merged**